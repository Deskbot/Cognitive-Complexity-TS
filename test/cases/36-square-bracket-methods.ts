var recursive;

class C {
    [Symbol.iterator]() {
        if (true) {

        }
    }

    [recursive]() {
        this[recursive]();
    }
}
