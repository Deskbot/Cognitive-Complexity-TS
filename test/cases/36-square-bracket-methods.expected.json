{
    "36-square-bracket-methods.ts": {
        "score": 2,
        "inner": [
            {
                "name": "C",
                "score": 2,
                "line": 3,
                "column": 1,
                "inner": [
                    {
                        "name": "[Symbol.iterator]",
                        "score": 1,
                        "line": 4,
                        "column": 5
                    },
                    {
                        "name": "[recursive]",
                        "score": 1,
                        "line": 10,
                        "column": 5
                    }
                ]
            }
        ]
    }
}
